src/services/codeFixProvider.ts

```diff
@@ -1,40 +1,56 @@
 /* @internal */
 namespace ts {
-    export interface CodeFix {
+    export interface CodeFixRegistration {
         errorCodes: number[];
-        getCodeActions(context: CodeFixContext): CodeAction[] | undefined;
+        getCodeActions(context: CodeFixContext): CodeFix[] | undefined;
+        actionIds: string[];
+        getAllCodeActions(context: CodeFixAllContext): CodeActionAll;

```

Comment:
we should make this one optional

Reply:
Or make CodeFixRegisraction a union type

Reply:
not all codeFixes need to provide a list of applicable actionIds, and not all of them need to provide a getAllCodeActions feature and not all of them need to return an `actionId` in their output.